src/main/scala/com/microsoft/hyperspace/index/FileBasedSignatureProvider.scala

```diff
@@ -26,26 +26,27 @@ import com.microsoft.hyperspace.util.HashingUtils
  * [[FileBasedSignatureProvider]] provides the logical plan signature based on files in the
  * logical relation. File metadata, eg. size, modification time and path, of each file in the
  * FileIndex will be used to generate the signature.
+ * If the logical plan does not have any LogicalRelation operator, no signature is provided.
  */
 class FileBasedSignatureProvider extends LogicalPlanSignatureProvider {
 
   /**
    * Generate the signature of logical plan.
    *
    * @param logicalPlan logical plan of data frame.
-   * @return signature.
+   * @return signature, if the logical plan has some LogicalRelation operator(s); Otherwise None.
    */
-  def signature(logicalPlan: LogicalPlan): String = {
-    HashingUtils.md5Hex(fingerprintVisitor(logicalPlan))
+  def signature(logicalPlan: LogicalPlan): Option[String] = {
+    fingerprintVisitor(logicalPlan).map(HashingUtils.md5Hex(_))
```

Comment:
I think `_` is not required: `fingerprintVisitor(logicalPlan).map(HashingUtils.md5Hex)`?

Reply:
Fixed.